packages/SwingSet/src/liveslots/liveslots.js

```diff
@@ -600,10 +600,16 @@ function build(
     return result;
   }
 
-  function registerValue(slot, val) {
+  function registerValue(slot, val, valIsCohort) {
```

Comment:
Here are the notes we made on vref terminology:
```
 * A slot string (aka "vref") can take one of the forms:
 *
 *   T-N
 *   T+N
 *   o+N/I
 *   o+N/I:F

 * vrefs for exported virtual objects
 *   single-facet: vref = o+N/I
 *    (kind handle: o+9/N)
 *    //cohort ID: o+N/I
 *    slotToVal.get(vref) -> representative
 *    valToSlot.get(representative) -> vref
 *    vom.rc.${vref}
 *    vom.es.${vref} = 'n/s/r'

 *   multi-facet:  vref = o+N/I:F
 *    (kind handle: o+9/N)
 *    cohortID: o+N/I
 *    slotToVal.get(cohortID) -> cohort record
 *    valToSlot.get(representative) -> vref
 *    vom.rc.${cohortID}
 *    vom.es.${cohortID} = /[nsr]{numFacets}/

 * "cohortID" vs "cohortRef" 
 * slotToVal = Map< (CohortID | RemotableExportID | PromiseID | DeviceID) ,
                    WeakRef< CohortRecord | Representative | Far | Presence | Promise | DeviceNode > >
 * convertSlotToVal(vref => Representative | Far | Presence | Promise | DeviceNode)
 * valToSlot = WeakMap<Representative | Far | Presence | Promise | DeviceNode, vref>
 * convertValToSlot(... => vref)

 * or CohortID includes o-N , p+N, d-N, etc, just not exactly virtual object vrefs
```

We decided that `slot` is the most generic term, used in the definition of CapData and things like a Message (with `target`, `method`, `args: CapData = { body, slots }`, and `result`). Within liveslots.js, `vref` is the most generic term. `convertSlotToVal` takes a vref, but the `slotToVal` table (which yes wants to be renamed) is actually indexed by something slightly different, since when the vref is really an exported virtual object (or maybe an exported multi-facet virtual object), the table is indexed by the cohortID rather than the vref.

As discussed, please add a comment describing our grand naming plan, maybe at the top of liveslots, or the top of `parseVatSlots`, to remind us about incremental improvements we can make over time.

Reply:
In the course of writing the requested comment in `parseVatSlots.js` I think I have clarified the terminology, and in a way that addresses the concern I expressed in my comment above.   For now I'm introducing with the term "baseRef", which we can replace with something better if we decide we don't like it, but which captures the distinction I'd like to express.

 A "vref" identifies an entity visible to vat code to which messages may be sent and which may be compared for equality to other such entities.  Let's call such an entity an "addressable object".
 
A "baseRef" designates an entity that is managed by LiveSlots, both as a unit of garbage collection specifically and as a unit of memory management more generally.  Such an entity may be a promise or remotable object or imported presence, all of which will always be JavaScript objects in memory, or it may be a virtual object or collection, which can be in memory or on disk or both.  Let's call such an entity a "base object".  In most cases this is one and the same with the addressable object that the vref designates, but in the case of a faceted object it is the faceted object as a whole (represented in memory, though not on disk, as the cohort array) rather than any particular individual facet (the faceted object per se is never exposed directly to code running within the vat; only its facets are).

Given these definitions, we can describe our various translation functions/tables:

`slotToVal` maps a baseRef to a base object (actually, to a weakRef that points to the base object)
`getValForSlot` maps a baseRef to a base object, or to undefined if it is not resident in memory
`convertSlotToVal` maps a vref to to an addressable object, loading it from disk if necessary
 
`valToSlot` maps an addressable object to a vref
`getSlotForVal` maps an addressable object to a vref
`convertValToSlot` maps an addressable object to a vref, generating a new vref if necessary

If we accept these as correct, some renaming of functions seems called for, probably best done in a separate PR.  In the meantime, I will rename the parameters and variables within them to reflect the asymmetric typing that the above definitions reveal.